src/corehost/cli/args.cpp

```diff
@@ -53,6 +53,16 @@ void setup_shared_store_paths(const hostpolicy_init_t& init, const pal::string_t
         args->dotnet_shared_store = own_dir;
         append_path(&args->dotnet_shared_store, RUNTIME_STORE_DIRECTORY_NAME);
         append_path(&args->dotnet_shared_store, init.tfm.c_str());
+
+        //Also add an architecture specific path to allow for unzipped developer cli experience
+        //where developer points runtime store operation output to "dotnet.exe" relative shared store folder
+
+        pal::string_t local_developer_store = own_dir ;
+        append_path(&local_developer_store, RUNTIME_STORE_DIRECTORY_NAME);
+        append_path(&local_developer_store, get_arch());
+        append_path(&local_developer_store, init.tfm.c_str());
+        local_developer_store.push_back(PATH_SEPARATOR);
+        (args->env_shared_store).push_back(local_developer_store);
```

Comment:
One thing I dislike about this approach is that it really interweaves where we are looking in a fashion that I don't think makes sense:

o %DOTNET_SHARED_STORE% -- multiple delimited paths
o dotnet.exe relative shared store - with architecture appended     `<-- you are adding this`
o $HOME/.dotnet/{x86|x64}/ or %USERPROFILE%\.dotnet\{x86|x64}
o dotnet.exe relative shared store - no architecture
o Global location